x-pack/plugins/fleet/server/services/epm/archive/parse.ts

```diff
@@ -182,23 +182,29 @@ export function parseAndVerifyArchive(
   const toplevelDir = topLevelDirOverride || paths[0].split('/')[0];
   paths.forEach((filePath) => {
     if (!filePath.startsWith(toplevelDir)) {
-      throw new PackageInvalidArchiveError('Package contains more than one top-level directory.');
+      throw new PackageInvalidArchiveError(
+        `Package at top-level directory ${toplevelDir} contains more than one top-level directory.`
```

Comment:
Could we log out the `filePath` as well? I've seen this error happening when the `toplevelDir` didn't contain the package name.